### PR TITLE
[sdk/javascript]: add caching of configuration files

### PR DESCRIPTION
**Pull request by Jaguar0625**
problem: network routes reload and reprocess server config files each GET
solution: cache parsed results

### CODE REVIEW

File: client/rest/test/catapult-sdk/utils/CachedFileLoader_spec.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+			const loader = new CachedFileLoader();
+			const processor = createSplitProcessor();
+
+			// Act:
+			return loader[functionName](tempFile.name, processor).then(() => {
+				// - slight pause before writing to make sure mtime is changed
+				const start = new Date().getTime();
+				while (new Date().getTime() <= start + 10)
+					;
```
Comment (AnthonyLaw) [inline client/rest/test/catapult-sdk/utils/CachedFileLoader_spec.js:109]: wondering can we use `setTimeout` here?

File: client/rest/test/routes/networkRoutes_spec.js [hunk 2]
```diff
@@ -66,47 +67,12 @@ describe('network routes', () => {
...
-					expect(mockServer.next.calledOnce).to.equal(true);
-					expect(mockServer.send.firstCall.args[0]).to.deep.equal({
-						network: { identifier: 'testnet' },
-						chain: { enableVerifiableState: true },
-						plugins: { aggregate: { maxTransactionsPerAggregate: '1\'000' } }
-					});
-					readFileStub.restore();
-				});
-			});
+			const assertCanRetrieveNetworkProperties = lines => {
```
Comment (AnthonyLaw) [inline client/rest/test/routes/networkRoutes_spec.js:70]: missing Arrange, Act, Assert comment.

File: client/rest/test/routes/networkRoutes_spec.js [hunk 3]
```diff
@@ -116,65 +82,62 @@ describe('network routes', () => {
...
+				'identifier = testnet',
+				'[chain]',
+				'enableVerifiableState = true',
+				'[private]',
+				'secretCode = 42',
+				'[plugin:catapult.plugins.aggregate]',
+				'maxTransactionsPerAggregate = 1\'000'
+			]));
+
+			it('fails when file does not exist', () => {
```
Comment (AnthonyLaw) [inline client/rest/test/routes/networkRoutes_spec.js:108]: missing Arrange, Act, Assert comment.

File: client/rest/test/routes/networkRoutes_spec.js [hunk 4]
```diff
@@ -269,39 +217,13 @@ describe('network routes', () => {
...
-					expect(mockServer.send.firstCall.args[0].message).to.equal('there was an error reading the network properties file');
-				});
-			});
-
-			it('errors when the file has an invalid format', () => {
-				readFileStub = sinon.stub(fs, 'readFile').callsFake((path, data, callback) =>
-					callback(null, '{ "not": "iniFormat" }'));
-
-				const services = { config: { apiNode: {} } };
+			it('fails when file does not exist', () => {
```
Comment (AnthonyLaw) [inline client/rest/test/routes/networkRoutes_spec.js:220]: missing Arrange, Act, Assert comment.

**Review by AnthonyLaw — COMMENTED** [review]
Excellent work

**Review by AnthonyLaw — APPROVED** [review]
Look good! 👍🏼

**Review by gimre-xymcity — APPROVED** [review]

**Pull request merged**